cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/SourceControlManagementService.java

```diff
@@ -153,17 +153,18 @@ public void validateRepository(NamespaceId namespace, RepositoryConfig repoConfi
 
   /**
    * The method to push an application to linked repository
-   * @param appRef {@link ApplicationReference}
+   *
+   * @param appRef        {@link ApplicationReference}
    * @param commitMessage optional commit message from user
    * @return {@link PushAppResponse}
-   * @throws NotFoundException if the application is not found or the repository config is not found
-   * @throws IOException if {@link ApplicationLifecycleService} fails to get the adminOwner store
-   * @throws PushFailureException if {@link SourceControlOperationRunner} fails to push
+   * @throws NotFoundException        if the application is not found or the repository config is not found
+   * @throws IOException              if {@link ApplicationLifecycleService} fails to get the adminOwner store
+   * @throws SourceControlException     if {@link SourceControlOperationRunner} fails to push
+   * @throws AuthenticationConfigException if the repository configuration authentication fails
    * @throws NoChangesToPushException if there's no change of the application between namespace and linked repository
    */
   public PushAppResponse pushApp(ApplicationReference appRef, @Nullable String commitMessage)
-    throws NotFoundException, IOException, PushFailureException,
-           NoChangesToPushException, AuthenticationConfigException {
+    throws NotFoundException, IOException, NoChangesToPushException, AuthenticationConfigException {
```

Comment:
Do we also need to specify throws `SourceControlException`?

Reply:
Not needed as `SourceControlException` is unchecked